src/index.ts

```diff
@@ -51,7 +51,7 @@ const calculate = (
   // Because of the rounding, even if the calculation is correct, sometimes there is €0.01 left over.
   // For example, if total is 31, divided by 3 it would be 10 / 10 / 10.
   // Also, there's a case, such as 29, where it returns negative result.
-  if (newEndAmount <= 1) {
+  if (depreciationAmount / previousEndAmount >= 0.99) {
```

Comment:
Could you add some tests to verify the fix?

Reply:
Yes, Sure.

Reply:
I've added 3 test, those if run with previous code, return none zero amount in last part, but in new code, all of them passed successfully.